homeassistant/components/drop_connect/sensor.py

```diff
@@ -0,0 +1,314 @@
+    entities: list[SensorEntity] = []
+
+    if device_type == DEV_HUB:
+        entities.extend(
+            DROPSensor(coordinator, sensor)
+            for sensor in SENSORS
+            if sensor.key
+            in (
+                AVERAGE_WATER_USED,
+                BATTERY,
+                CURRENT_FLOW_RATE,
+                CURRENT_SYSTEM_PRESSURE,
+                HIGH_SYSTEM_PRESSURE,
+                LOW_SYSTEM_PRESSURE,
+                PEAK_FLOW_RATE,
+                WATER_USED_TODAY,
+            )
+        )
+    elif device_type == DEV_SOFTENER:
+        entities.extend(
+            DROPSensor(coordinator, sensor)
+            for sensor in SENSORS
+            if sensor.key
+            in (
+                BATTERY,
+                CAPACITY_REMAINING,
+                CURRENT_FLOW_RATE,
+                CURRENT_SYSTEM_PRESSURE,
+            )
+        )
+    elif device_type == DEV_FILTER:
+        entities.extend(
+            DROPSensor(coordinator, sensor)
+            for sensor in SENSORS
+            if sensor.key
+            in (
+                BATTERY,
+                CURRENT_FLOW_RATE,
+                CURRENT_SYSTEM_PRESSURE,
+            )
+        )
+    elif device_type == DEV_LEAK_DETECTOR:
+        entities.extend(
+            DROPSensor(coordinator, sensor)
+            for sensor in SENSORS
+            if sensor.key
+            in (
+                BATTERY,
+                TEMPERATURE,
+            )
+        )
+    elif device_type == DEV_PROTECTION_VALVE:
+        entities.extend(
+            DROPSensor(coordinator, sensor)
+            for sensor in SENSORS
+            if sensor.key
+            in (
+                BATTERY,
+                CURRENT_FLOW_RATE,
+                CURRENT_SYSTEM_PRESSURE,
+                TEMPERATURE,
+            )
+        )
+    elif device_type == DEV_PUMP_CONTROLLER:
+        entities.extend(
+            DROPSensor(coordinator, sensor)
+            for sensor in SENSORS
+            if sensor.key
+            in (
+                CURRENT_FLOW_RATE,
+                CURRENT_SYSTEM_PRESSURE,
+                TEMPERATURE,
+            )
+        )
+    elif device_type == DEV_RO_FILTER:
+        entities.extend(
+            DROPSensor(coordinator, sensor)
+            for sensor in SENSORS
+            if sensor.key
+            in (
+                CARTRIDGE_1_LIFE,
+                CARTRIDGE_2_LIFE,
+                CARTRIDGE_3_LIFE,
+                INLET_TDS,
+                OUTLET_TDS,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class DROPSensor(DROPEntity, SensorEntity):
+    """Representation of a DROP sensor."""
+
+    entity_description: DROPSensorEntityDescription
+
+    def __init__(self, device, entity_description) -> None:
+        """Initialize the sensor."""
+        super().__init__(entity_description.key, device)
+        self.entity_description = entity_description
+        self.device = device
```

Comment:
If I remove this definition then it breaks the `native_value()` function below, which references it.

Reply:
first i think we should rename the variable to what it is:
```diff
-def __init__(self, device, entity_description) -> None:
+def __init__(self, coordinator, entity_description) -> None:
```

than we should also add typing in here
```diff
-def __init__(self, coordinator, entity_description) -> None:
+def __init__(self, coordinator: DROPDeviceDataUpdateCoordinator, entity_description: DROPSensorEntityDescription) -> None:
```

now just adjust the `native_value()` to use `self.coordinator` instead of `self.device`

Reply:
Fixed in 89ef7d6e2a9c9ab7e542ff7f719ac56930b5c361. Thanks for your patience through this review; I'm learning...

Reply:
you're welcome and thanks to not give up 🙂 i think we are close to be ready👍